src/Claim.ts

```diff
@@ -1,14 +1,72 @@
 /* tslint:disable:no-relative-imports */
-import * as bitcore from 'bitcore-lib'
 import * as crypto from 'crypto'
-import { canonize } from 'jsonld'
+import * as cuid from 'cuid'
+import * as jsonld from 'jsonld'
+import * as jsig from 'jsonld-signatures'
 
 import { IllegalArgumentException } from './Exceptions'
-import { Claim, ClaimAttributes, ClaimType, ClaimContext, isClaim } from './Interfaces'
+import {
+  Claim,
+  ClaimAttributes,
+  ClaimType,
+  ClaimContext,
+  DefaultClaimContext,
+  claimTypeDefaults,
+  isClaim,
+} from './Interfaces'
+import { DataParser } from './util/DataParser'
 
-export const canonizeClaim = async (claim: Claim): Promise<string> => {
-  const contextualClaim = { ...ClaimContext, ...claim }
-  return canonize(contextualClaim)
+DataParser(jsonld)
+jsig.use('jsonld', jsonld)
+
+export const canonizeClaim = async (document: Claim): Promise<string> => {
+  const contextualClaim = {
+    '@context': {
+      ...DefaultClaimContext['@context'],
+      ...claimTypeDefaults[document.type]['@context'],
+      ...document['@context'],
+    },
+    type: document.type,
+    issuer: document.issuer,
+    issuanceDate: document.issuanceDate,
+    claim: document.claim,
+  }
+  return jsonld.canonize(contextualClaim)
+}
+
+/*
+  signClaim = (signingOptions: any) => async (document: Claim): Promise<Claim>
+    signingOptions:
+      algorithm: 'Ed25519VerificationKey2018'
+      creator: 'data:,<base58PublicKeyValue>'
+      privateKeyBase58: base58 rendition of an Ed22159 private key
+    document: any valid Claim
+  NOTE: If you choose to use a different signing algorhithm, the other signingOption keys may differ.
+  Refer to the jsonld-signatures library for more detail
+*/
+export const signClaim = (signingOptions: any) => async (document: Claim): Promise<Claim> => {
```

Comment:
What do you think about moving signClaim to its own file? Currently to unit test the function we are exporting it which means its also being exported as part of the [public API](https://github.com/poetapp/poet-js/blob/master/src/index.ts). I don't think we want to expose it to the public API so moving it to another file would prevent that and still allow it to be unit tested. I think we could also move the jsig.verify logic their as well?

Reply:
Same with canonizeClaim actually, not sure if we meant to expose those as part of the public api or not.

Reply:
If we move it to another file, don't we have to export it anyways? Part of the `createClaim` requires signing the file and part of `isValidClaim` requires `isValidSignature`.

Reply:
Take a look at `src/index.js` it exports everything that `src/Claim.js` exports. If we move `signClaim` to another file and import it into `src/Claim.js` we don't need to export it from `src/Claim.js'

Reply:
I pinged @lautarodragan so we can get some input from him as well.

Reply:
Just catching up with this so I may be missing something, but what's the issue with exporting `signClaim`? Don't we want users to be able to sign claims themselves, even if they choose not to use our `createClaim`? 

(Can't think of a reason why they'd want that, but is there any evil done in providing them the option?)

Reply:
No issue with exposing it if that is what we want to do and think someone has a use case. I was thinking it was a detail of createClaim that shouldn't be exposed to prevent confusion and to keep the public api minimal and simple.

Reply:
However I think Kenny raises a good issue about `cannonizeClaim`. I can't think of a use case here.

Reply:
I probably should extract the tests around `cannonizeClaim`, and simply focus on `getClaimId`

Reply:
Sounds good on cannonizeClaim, @krobi64 want to create a new issue for that real quick?

Reply:
oops I already did it :P

Reply:
and I see you went and split the tests up haha

Reply:
It helped me organize the tests better. I noticed some tests were scattered throughout the file.

Reply:
Sounds like we want to leave `signClaim` as part of the API. I will resolve this for now.

Reply:
Thanks @kennylavender, now I understand the issue.

I don't have a strong opinion for or against, whatever you guys decide is fine. 

The only thing I can think of is: if we leave it in the public API and then want to remove it from it, that'd be a breaking change. The other way around would just be a major version bump, but no breaking change.